### ISSUE TITLE
Unable to use network_profile in azurerm_kubernetes_cluster

### ISSUE THREAD

**Issue by engrasif** [terraform-providers/terraform-provider-azurerm#1716]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

v0.11.7

### Affected Resource(s)

* azurerm_kubernetes_cluster

### Terraform Configuration Files

Unable to user network_profile in azurerm_kubernetes_cluster

```hcl
resource "azurerm_kubernetes_cluster" "test" {
  name                = "example"
  location            = "West EU"
  resource_group_name = "example-rg"
  dns_prefix          = "test"
  kubernetes_version  = "1.10.3"

  linux_profile {
    admin_username = "testuser"

    ssh_key {
      key_data = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCqaZoyiz1qbdOQ8xEf6uEu1cCwYowo5FHtsBhqLoDnnp7KUTEBN+L2NxRIfQ781rxV6Iq5jSav6b2Q8z5KiseOlvKA/RF2wqU0UPYqQviQhLmW6THTpmrv/YkUCuzxDpsH7DUDhZcwySLKVVe0Qm3+5N2Ta6UYH3lsDf9R9wTP2K/+vAnflKebuypNlmocIvakFWoZda18FOmsOoIVXQ8HWFNCuw9ZCunMSN62QGamCe3dL5cXlkgHYv7ekJE15IA9aOJcM7e90oeTqo+7HTcWfdu0qQqPWY5ujyMw/llas8tsXY85LFqRnr3gJ02bAscjc477+X+j/gkpFoN1QEmt terraform@demo.tld"
    }
  }

  agent_pool_profile {
    name           = "default"
    count          = "2"
    vm_size        = "Standard_DS2_v2"
    vnet_subnet_id = "${azurerm_subnet.test.id}"
  }

  service_principal {
    client_id     = "################"
    client_secret = "################"
  }

  network_profile {
    network_plugin     = "azure"
    dns_service_ip     = "10.0.0.10"
    docker_bridge_cidr = "172.17.0.1/16"
    service_cidr       = "10.0.0.0/16"
  }
}
```

### Debug Output

"terrafrom plan" shows below error

Error: module.compute.module.kubernetes_cluster.azurerm_kubernetes_cluster.cluster: invalid or unknown key: network_profile

### Actual Behavior

Error: module.compute.module.kubernetes_cluster.azurerm_kubernetes_cluster.cluster: invalid or unknown key: network_profile

### Steps to Reproduce

1. `terraform plan`

**Comment by tombuildsstuff** [comment]
hi @engrasif 

Thanks for opening this issue :)

Support for Network Profiles within AKS landed in v1.12.0 of the AzureRM Provider which was released about 10 minutes after this issue was opened; would you be able to upgrade and take a look? You can do this via:

```
provider "azurerm" {
  version = "=1.12.0"
}
```

and then re-running `terraform init`.

Since this is a question regarding Terraform Configuration rather than a bug in Terraform - I'm going to close this issue for the moment.

Thanks!

**Comment by engrasif** [comment]
thanks :), i just tested and its working fine.
